enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,493 @@
+
+![alt_text](./assets/monitoring-stack-prometheus-baseline.png)
+
+At the latest timestamp, the total memory usage for the 3 instances forming functional sharding is 434MB, while the memory usage of the single instance is 316MB. What is also worth pointing out, and is already illustrated by the graph, is that systems that do not export many metrics, such as node_exporter and kubelet, require small Prometheus instances in terms of memory requirements. This leads to easier scheduling of Prometheus pods across a wider range of nodes.
+
+We also performed some back-of-the-envelope calculations in order to illustrate the added cost of functional sharding in terms of actual money. A `c5.xlarge` on-demand instance in AWS (a compute optimized instance with 4 CPUs and 8GB RAM) costs 140$ per month or 18$ per GB per month.
+When using reserved capacity, the cost goes down to 11$ per GB per month for the same instance with a 1 year reservation.
+
+Our conclusion based on these factors is as follows: while sharing a Prometheus instance does come with better resource utilization, we do not believe it justifies the reduced resiliency that it also brings along.
+The resiliency of such a shared instance would be further reduced as more managed services are installed in a cluster. In addition, a shared Prometheus instance is often more volatile regarding resource utilization, which leads to users over provisioning resources to prevent OOMs by misconfiguration or complex querying patterns.
+On the other hand, an independent Prometheus instance for each add-on yields a better separation of concerns by allowing autonomy, control and responsibility to be delegated to the appropriate teams. In other words, addon owners will become fully responsible for properly configuring and sizing their Prometheus instance to support the query patterns they need and the time series they need to store.
+Finally, with the proposed stack we nevertheless retain the option of deploying multi-namespace monitoring stacks. This will effectively support the pattern of sharing a Prometheus instance between managed services.
+
+### Migration
+
+There are multiple monitoring solutions that Addons currently use:
+
+* AppSRE have their monitoring stack for SaaS services, which they plan to migrate to UWM
+  * MSO is planned to have feature parity with UWM with the exception of global view with Platform Monitoring. We will initially use federation to close this gap (See )
+* Platform Monitoring
+  * The PM stack cannot be used for user/managed service monitoring anyway. Therefore no migration is predicted so far.
+* Observability Operator
+  * This is great work done by the Kafka team to get unblocked and as we see it has feature parity to UWM with addition of Dashboarding and Promtail (logging aggregation). MSO will provide all but logging integration. For the logging part, Openshift Logging Operator should be leveraged.
+
```

Comment:
I don't see a mention of the ODF-MS monitoring stack solution, which to me hints that we never verified feature parity with that solution. It is similar to the "Observability Operator" but does differ on some points

Reply:
I think we discussed your requirements e.g in ODF Managed Service SD sync, so I don't think you are forgotten. We addressed all your questions in this proposal too. *Did we miss anything?*

We don't have concrete clarity on your current ODF-MS monitoring stack solution. As far I as I remember it was based on PM (Platform Monitoring) which was primarily not designed for this work. Is there any extra functionality of ODF-MS that is worth mentioning here?

Reply:
ODF-MS monitoring stack is not based on PM. I think the main thing that is missing is around "Monitoring Resource" selection and OCP console integration. I do believe I have left separate comments addressing these concerns.

Reply:
perect, thanks.